MaxText/layers/linears.py

```diff
@@ -404,27 +417,56 @@ def gmm(inputs, kernel, group_sizes):
 
       inputs = inputs.astype(self.dtype)
       kernel = kernel.astype(self.dtype)
+
+      quantization_config: str = self.config.quantization
```

Comment:
Will this look a little bit better?

```
quantization_config = self.config.quantization

quantization_types = {
    "int8": (jnp.int8, jnp.int8),
    "int8w": (None, jnp.int8),
    "int4w": (None, jnp.int4),
}

lhs_quantize_dtype, rhs_quantize_dtype = quantization_types.get(
    quantization_config, (None, None)
)
```

Also could you help add a coment for int8w and int4w [here](https://github.com/AI-Hypercomputer/maxtext/blob/58048291bb380474111b18166d54f413071fc7bd/MaxText/configs/base.yml#L73-L75)? Thank you!

Reply:
Sounds better, I've added more logic to avoid conflict with unsupported quantization configurations.